api/v1alpha1/virtualmachine_types.go

```diff
@@ -102,9 +102,27 @@ type VirtualMachineVolume struct {
 	// have a unique name.
 	Name string `json:"name"`
 
+	// StoragePolicy associated with volume.
+	StoragePolicy   *string `json:"storagePolicy,omitempty"`
```

Comment:
StoragePolicy will only apply to some types of volumes. For example, it won't apply to PersistentVolumeClaims (if there is a storage policy, it would be defined with the PV itself). 

This field probably can be moved in to the VSphereVolumeSource struct. Otherwise, please expand the documentation to state the limitations.

Reply:
Per Derek's comment in the design doc, we should specify this at the VirtualMachineVolume level. As you rightly said, for PersistentVolumeClaims (if there is a storage policy, it would be defined with the PV itself). I'll update my comment to say it only current applies to the VSphereVolumeSource type. I think the reasoning behind this was to avoid duplicating that field in all the different volume types. We need to decide the override behavior for PVC however. I am assuming we don't want this to override the one in the PV spec